### PR TITLE
add g8scontrolplane validation

### PR DESCRIPTION
**Pull request by anvddriesch**

### CODE REVIEW

**Review by njuettner — APPROVED** [review]
Cool 🚀, when you have time adding a test would be awesome. But I know it's in progress 🙂

File: pkg/unittest/default_admission_request.go [hunk 1]
```diff
@@ -33,3 +33,27 @@ func DefaultAdmissionRequestAWSControlPlane() (v1beta1.AdmissionRequest, error)
...
+func DefaultAdmissionRequestG8sControlPlane() (v1beta1.AdmissionRequest, error) {
+	byt, err := json.Marshal(DefaultG8sControlPlane())
+	if err != nil {
+		return v1beta1.AdmissionRequest{}, microerror.Mask(err)
+	}
+
+	req := v1beta1.AdmissionRequest{
+		Kind: metav1.GroupVersionKind{
+			Version: "infrastructure.giantswarm.io/v1alpha2",
+			Kind:    "AWSControlPlane",
```
Comment (njuettner) [inline pkg/unittest/default_admission_request.go:46]:
```suggestion
			Kind:    "G8sControlPlane",
```

File: pkg/unittest/default_admission_request.go [hunk 2]
```diff
@@ -33,3 +33,27 @@ func DefaultAdmissionRequestAWSControlPlane() (v1beta1.AdmissionRequest, error)
...
+		},
+		Resource: metav1.GroupVersionResource{
+			Version:  "infrastructure.giantswarm.io/v1alpha2",
+			Resource: "awscontrolplanes",
```
Comment (njuettner) [inline pkg/unittest/default_admission_request.go:50]:
```suggestion
			Resource: "g8scontrolplanes",
```

**Review by njuettner — CHANGES_REQUESTED** [review]
You need to add webhook config for g8scontrolplane, see https://github.com/giantswarm/aws-admission-controller/blob/master/helm/aws-admission-controller/templates/webhook.yaml#L94

File: pkg/aws/g8scontrolplane/validate_g8scontrolplane.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+			return nil
+		}
+	}
+
+	{
+		b := backoff.NewMaxRetries(3, 1*time.Second)
+		err = backoff.Retry(fetch, b)
+		// Note that while we do log the error, we don't fail if the AWSControlPlane doesn't exist yet. That is okay because the order of CR creation can vary.
+		if aws.IsNotFound(err) {
+			v.Log("level", "debug", "message", fmt.Sprintf("No AWSControlPlane %s could be found: %v", g8sControlPlane.GetName(), err))
```
Comment (njuettner) [inline pkg/aws/g8scontrolplane/validate_g8scontrolplane.go:90]: Maybe also add the cluster id, wdyt?

**Comment by anvddriesch** [comment]
I added the webhook config and tested this on `giraffe`. It's working.
```
Error from server: error when creating "testg8scp.yaml": admission webhook "g8scontrolplanes.aws-admission-controller-unique.giantswarm.io" denied the request: not allowed error: G8sControlPlane gcat9 with 1 replicas does not match AWSControlPlane gcat9 with 3 availability zones [cn-north-1a cn-north-1a cn-north-1b]
```

**Review by njuettner — APPROVED** [review]
cool 👍

**Pull request merged**